home_audit.node.js

```diff
@@ -878,7 +878,13 @@ let validationRules = {
         return this._heating_year(value);
     },
     _heating_year: function(value) {
-        return new Validation(TypeRules._int(value, parseInt(_homeValues.year_built), (new Date()).getFullYear()), BLOCKER);
+        let isNotApiSafe = new Validation(TypeRules._int(value, 1970, (new Date()).getFullYear()), BLOCKER);
+        if(!isNotApiSafe) {
+            //Check if fits bounds for home
+            new Validation(TypeRules._int(value, parseInt(_homeValues.year_built), (new Date()).getFullYear()), ERROR);
```

Comment:
I really hate being that guy, but... did you test this? Like, did you enter a value that was before the house was built but after 1970 and confirm the type of the resulting validation message? There are some problems here that I have to imagine would have become evident during such a test:

- You're not actually returning the Validation you create in line 884, you're just creating an instance of Validation and then throwing it away.
- Line 884 can't actually ever be reached, because `!isNotApiSafe` is always `false` because `isNotApiSafe` is an instance of `Validation`, and class instances in JavaScript can't ever be falsey. The result is that every value that isn't a blocker is going to be treated as valid.

Please make sure this works as expected and make sure there's a functional test covering it. Thank you!

Reply:
You gotta be "that guy."  I did, but lazily.  I just manually made sure the validation was still showing, but I was dumb and forgot the point was that it wasn't a BLOCKER, so it looked good to me.  Added a return, and will run my falsey check on the message and not the class itself.  Sorry for the oversight! I'll write a test that makes sure we save successfully within the appropriate bounds.

Reply:
<3

Is there any way in the frontend to distinguish between the various error types? I feel like there should be some visible signal that some error is the one that's preventing the label from being submitted - which would also give you a hook for testing that an error is a blocker and not just a regular error.

Reply:
I've actually thought of that too.  It would be pretty easy.  I'd add a class that matches our error type to invalid_feedback that gives a colored border that matches our icons in the headers (ie, BLOCKER: red, ERROR: yellow, MANDATORY: grey).

Reply:
Let's talk about this on Monday. There are a few things I want to consider before we actually draw up a feature ticket for this.